lib/internal/worker.js

```diff
@@ -79,6 +79,23 @@ class Worker extends EventEmitter {
       }
     }
 
+    let env;
+    if (typeof options.env === 'object' && options.env !== null) {
+      env = Object.create(null);
+      for (const [ key, value ] of Object.entries(options.env))
+        env[key] = `${value}`;
```

Comment:
This is not inheriting process.env when the option is passed?

Reply:
Yes, if you pass `options.env` explicitly, it won’t inherit from `process.env`, like we do for child processes